controllers/awsmachine_controller.go

```diff
@@ -411,7 +411,12 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	}
 
 	// Make sure Spec.ProviderID is always set.
-	machineScope.SetProviderID(fmt.Sprintf("aws:////%s", instance.ID))
+	failureDomain := machineScope.AWSMachine.Spec.FailureDomain
+	if failureDomain == nil {
```

Comment:
we should use a `deref` helper function,
```
if failureDomain is nil it returns ""
else return * failureDomain
```
Then we can just
`machineScope.SetProviderID(fmt.Sprintf("aws:///%s/%s", failureDomain, instance.ID))`

Reply:
👍